src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -276,6 +276,27 @@ protected override ImmutableArray<PendingBranch> RemoveReturns()
             return result;
         }
 
+        private static bool HasAwait(PendingBranch pending)
+        {
+            if (pending.Branch is null)
+            {
+                return false;
+            }
+
+            BoundKind kind = pending.Branch.Kind;
+            switch (kind)
+            {
+                case BoundKind.AwaitExpression:
+                    return true;
+                case BoundKind.UsingStatement:
+                    var usingStatement = (BoundUsingStatement)pending.Branch;
+                    var usingSyntax = (UsingStatementSyntax)usingStatement.Syntax;
```

Comment:
>var usingSyntax = (UsingStatementSyntax)usingStatement.Syntax; [](start = 20, length = 62)

I think it would be better to not rely on syntax, but on information from bound node only. We should capture the kind of using inside the bound node unambiguously.

Reply:
I had to rely on syntax because I was bailing out during binding in error scenarios. Now that binding perseveres more, I'll try alternative approach again.

Reply:
> I had to rely on syntax because I was bailing out during binding in error scenarios.

Even when minding bails, it should still capture all relevant semantic information in bound nodes.

---
In reply to: [161335806](https://github.com/dotnet/roslyn/pull/23961#discussion_r161335806) [](ancestors = 161335806)